### ISSUE TITLE
OIDC check_session_iframe does not parse origin correctly when path is included in referer

### ISSUE THREAD

**Issue by sasaken555** [OpenLiberty/open-liberty#25351]
**Describe the bug**  
When I check SSO session status with OpenID Connect provider's session management endpoint, iframe content (opiframe.js by session management endpoint) cannot complete to check request every check loop.

Browser console log: `Unable to complete request from https://client.example.com:9444/`

**Steps to Reproduce**  
1. Start OP (https://openidprovider.example.com:9443) with server.xml below.
2. Setup OIDC client application (https://client.example.com:9444) that loads scripts to calls session_management_endpoint, based on example in ["Invoking the Session Management Endpoint for OpenID Connect"](https://www.ibm.com/docs/en/was-liberty/base?topic=liberty-invoking-session-management-endpoint-openid-connect#tasktwlp_oidc_session_mgmt_endpoint__example__1) page.
3. Access client to `https://client.example.com:9444/` , and authenticate with OIDC.
4. After authentication succeeded, session check script loaded.
5. Browser console reports "Unable to complete request client.example.com:9444 [opiframe.js:70:11]".

**Expected behavior**  
iframe content (opiframe.js and sessionMgmt.jsp) should parse and extract origin correctly, then complete to request to session management endpoint.

Session management endpoint first returns sessionMgmt.jsp.
This JSP parses origin from HTTP referer, but parsing origin may not consider `/` in scheme when referer includes path. This causes invalid `EXPECTED_ORIGIN`.
For example in steps above, `https://openidprovider.example.com:9443/` will be `https:`.
https://github.com/OpenLiberty/open-liberty/blob/integration/dev/com.ibm.ws.security.openidconnect.server/resources/sessionMgmt.jsp#L30

```java
String expectedRpOrigin = request.getHeader("referer");
if (expectedRpOrigin != null) {
   	URL referrerUrl = new URL(expectedRpOrigin);
   	String path = referrerUrl.getPath();
   	if (path != null && !path.isEmpty()) {
                // When path is present, splitting with slash does not work expected!!
   		expectedRpOrigin = expectedRpOrigin.substring(0, expectedRpOrigin.indexOf(path));
   	}
}
```

When HTTP referer in Step4 is `https://openidprovider.example.com:9443/`, `EXPECTED_ORIGIN` in opiframe.js should be `https://openidprovider.example.com:9443` .

**Diagnostic information:**  
 - OpenLiberty Version: 23.0.0.5
 - Affected feature(s): openidConnectServer-1.0
 - Java Version: IBM Semeru Runtime Open Edition 17.0.6.0 (build 17.0.6+10)
 - server.xml configuration (WITHOUT sensitive information like passwords)

```xml
<server description="ID Provider server">

    <!-- Enable features -->
    <featureManager>
        <!-- OpenID Connect Provider -->
        <feature>openidConnectServer-1.0</feature>
        <feature>appSecurity-3.0</feature>
    </featureManager>

    <httpEndpoint id="defaultHttpEndpoint" httpPort="9080" httpsPort="9443" host="*" />
    <applicationManager autoExpand="true" />
    <webApplication contextRoot="/openidprovider" location="openidprovider.war" />

    <keyStore id="defaultKeyStore" password="***" />
    <ssl id="defaultSSLConfig" trustDefaultCerts="true" />

    <basicRegistry id="basic" realm="myRealm">
        <user name="alice" password="****" />
        <group name="my_group">
            <member name="alice" />
        </group>
    </basicRegistry>

    <openidConnectProvider id="OP"
                           sessionManaged="true" keyStoreRef="defaultKeyStore"
                           oauthProviderRef="oauth" signatureAlgorithm="RS256" jwkEnabled="true" />

    <oauthProvider id="oauth" httpsRequired="false" jwtAccessToken="true" tokenFormat="jwt">
        <localStore>
            <client name="sample-web" displayname="SampleWeb" enabled="true"
                   secret="***" redirect="https://client.example.com:9444/callback"/>
        </localStore>
    </oauthProvider>

    <oauth-roles>
        <authenticated>
            <special-subject type="ALL_AUTHENTICATED_USERS"/>
        </authenticated>
    </oauth-roles>
</server>
```

 - If it would be useful, upload the messages.log file found in `$WLP_OUTPUT_DIR/messages.log`

**Additional context**  
N/A

**Comment by sasaken555** [comment]
Maybe this issue is related to this comment.
https://github.com/OpenLiberty/open-liberty/issues/18105#issuecomment-897917081

**Comment by ayoho** [comment]
Thank you for opening an issue, @sasaken555. I've just delivered a PR that should fix the expected origin calculation, and the fix should be included in the next release of Open Liberty (23.0.0.6).
